src/main/java/br/com/sunty/config/AutenticacaoService.java

```diff
@@ -0,0 +1,30 @@
+package br.com.sunty.config;
+
+import br.com.sunty.models.Usuario;
+import br.com.sunty.repository.usuario.UsuarioRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class AutenticacaoService implements UserDetailsService {
+    private final UsuarioRepository usuarioRepository;
+
+    public AutenticacaoService(UsuarioRepository usuarioRepository) {
+        this.usuarioRepository = usuarioRepository;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        Optional<Usuario> usuario = usuarioRepository.findByEmail(username);
+
+        if(usuario.isPresent()){
+            return usuario.get();
+        }
+
+        throw new UsernameNotFoundException("Dados inválidos");
```

Comment:
Daria pra simplificar com aquela manhazinha:

```suggestion
        return usuarioRepository.findByEmail(username).orElseThrow(() -> new UsernameNotFoundException("Dados inválidos"));
```